nest/.gitlab-ci.yml

```diff
@@ -0,0 +1,124 @@
+stages:
+  - build
+  - test
+  - analyze
+  - deploy
+
+# Set image to be used across all stages
+image: node:16-alpine
+
+variables:
+  SONAR_PROJECT: $CI_PROJECT_NAME
+
+# Include common GitLab jobs used internally in Intility
+include:
+  - project: "developer-infrastructure/backstage/intility-software-templates"
+    file: "/ci-template/techdoc.gitlab-ci.yml"
+  - project: "soc/static-application-security"
+    file: "/StaticApplicationSecurity.gitlab-ci.yml"
+  - project: "digital-telco/utils/gitlabci"
+    file:
+      - sonarqube/SonarQube-Template.gitlab-ci.yml
+      - git/Gitleaks-Template.gitlab-ci.yml
+      - npm/npm-audit-Template.gitlab-ci.yml
+      - docker/Dockerfile-audit.gitlab-ci.yml
+
+# Install dependencies and compile typescript for later jobs
+build:
+  stage: build
+  artifacts:
+    paths:
+      - dist
+      - node_modules
+    expire_in: "1 day"
+  before_script:
+    - npm ci
+  script:
+    - npm run build
+  except:
+    - tags
+
+# Execute unit tests and upload test results to GitLab
+test:
+  stage: test
+  # Upload coverage percentage from terminal output to GitLab: https://docs.gitlab.com/ee/ci/yaml/#coverage
+  coverage: '/All files[^|]*\|[^|]*\s+([\d\.]+)/'
+  needs:
+    - job: build
+      artifacts: true
+  script:
+    - npm run test -- --ci --coverage
+  artifacts:
+    when: always
+    paths:
+      - coverage
+    # Upload coverage data ti GitLab: https://docs.gitlab.com/ee/ci/testing/test_coverage_visualization.html
+    reports:
+      coverage_report:
+        coverage_format: cobertura
+        path: coverage/cobertura-coverage.xml
+    expire_in: 1 day
+  except:
+    - tags
+
+# SonarQube code analysis upload.
+# This overwrites the default script provided by included job: https://gitlab.intility.com/digital-telco/utils/gitlabci/-/blob/master/sonarqube/SonarQube-Template.gitlab-ci.yml
+# TODO: Insert link to SonarQube docs
```

Comment:
This should be replaced with a link to the docs

Reply:
But we dont have a doc yet :D

Reply:
Never mind then 😄